src/cares.c

```diff
@@ -486,16 +489,34 @@ query_txt_cb(void *arg, int status,int timeouts, unsigned char *answer_buf, int
         goto callback;
     }
 
+    tmp_obj = NULL;
+    assembled_txt = NULL;
     for (txt_ptr = txt_reply; txt_ptr != NULL; txt_ptr = txt_ptr->next) {
-        tmp = PyStructSequence_New(&AresQueryTXTResultType);
-        if (tmp == NULL) {
+        // Initialzing new assembled record if necessary
+        if (txt_ptr->record_start == 1) {
+            tmp_obj = PyStructSequence_New(&AresQueryTXTResultType);
+            if (tmp_obj == NULL) {
+                break;
+            }
+            assembled_txt = PyBytes_FromString("");
+        }
+        // Constructing assembled record
+        assembled_ttl = txt_ptr->ttl;  // This will overwrite each time, but TTL should be the same for each chunk of a record
+        PyBytes_ConcatAndDel(&assembled_txt, PyBytes_FromString((char*)txt_ptr->txt));
+        if (assembled_txt == NULL) {
             break;
         }
-        PyStructSequence_SET_ITEM(tmp, 0, Py_BuildValue("s", (const char *)txt_ptr->txt));
-        PyStructSequence_SET_ITEM(tmp, 1, PyInt_FromLong((long)txt_ptr->ttl));
-        PyList_Append(dns_result, tmp);
-        Py_DECREF(tmp);
+        // Add previously assembled record to dns_result if necessary
+        if (txt_ptr->next == NULL || txt_ptr->next->record_start == 1) {
+            PyStructSequence_SET_ITEM(tmp_obj, 0, Py_BuildValue("s", PyBytes_AsString(assembled_txt)));
```

Comment:
I did this originally, but it created inconsistencies between Python 2 and 3. Without the Py_BuildValue, the result will be in 'bytes' in Python3 and Python2 would just be 'str'. What behaviour would be most desirable here?